bitnami/mongodb-sharded/templates/shard/shard-data-statefulset.yaml

```diff
@@ -40,7 +40,7 @@ spec:
       schedulerName: {{ $.Values.common.schedulerName | quote }}
       {{- end }}
       {{- if $.Values.shardsvr.dataNode.affinity }}
-      affinity: {{- include "common.tplvalues.render" (dict "value" $.Values.shardsvr.dataNode.affinity "context" $) | nindent 8 }}
+      affinity: {{- include "common.tplvalues.render" (dict "value" $.Values.shardsvr.dataNode.affinity "context" (set $ "dataNodeLoopId" $i)) | nindent 8 }}
```

Comment:
This is useless the `.Values.shardsvr.dataNode.affinity` parameter has the block below in it:
```yaml
              matchLabels:
                shard: "{{ .dataNodeLoopId }}"
```
This should be reflected in the **values.yaml**, currently it doesn't say anything about this, see:
```yaml
    ## Affinity for pod assignment
    ## ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity
    ## Note: shardsvr.dataNode.podAffinityPreset, shardsvr.dataNode.podAntiAffinityPreset, and shardsvr.dataNode.nodeAffinityPreset will be ignored when it's set
    ##
    affinity: {}
```

Reply:
```console
$ helm template . -s templates/shard/shard-data-statefulset.yaml -f values.yaml
  template:
    metadata:
      labels:
        app.kubernetes.io/name: mongodb-sharded
        helm.sh/chart: mongodb-sharded-3.2.9
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/component: shardsvr
        shard: "0"
    spec:
      affinity:
        matchLabels:
          shard: '0'
...
  template:
    metadata:
      labels:
        app.kubernetes.io/name: mongodb-sharded
        helm.sh/chart: mongodb-sharded-3.2.9
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/component: shardsvr
        shard: "1"
    spec:
      affinity:
        matchLabels:
          shard: '1'
```
just by setting
```yaml
affinity:
  matchLabels:
    shard: "{{ .dataNodeLoopId }}"
```
in the _values.yaml_